libV2/schemaUtils.js

```diff
@@ -1914,7 +1997,33 @@ let QUERYPARAM = 'query',
         return;
       }
 
-      let paramValue = resolveValueOfParameter(context, param);
+      let propertyDetails = {},
+        properties = {},
+        keyName,
+        paramValue = resolveValueOfParameter(context, param);
+
+      if (param && param.schema) {
+        const { name, schema } = param;
+        keyName = name;
+        properties = {
+          type: schema.type || 'unknown',
+          format: schema.format || undefined,
+          default: schema.default || undefined,
+          required: param.required || false,
+          deprecated: param.deprecated || false,
+          enum: schema.enum || undefined,
+          minLength: schema.minLength || undefined,
```

Comment:
I feel we don't need to add these attributes unless these are defined in the schema.
And same for header, pathParam and other params